commons/src/main/scala/it/unibo/scafi/space/Point.scala

```diff
@@ -22,6 +22,14 @@ class Point3D(val x: Double, val y: Double, val z: Double) extends Serializable
   }
 
   override def hashCode(): Int = Seq(x, y, z).hashCode()
+
+  override def _1: Double = x
+
+  override def _2: Double = y
+
+  override def _3: Double = z
+
+  override def canEqual(otherObject: Any): Boolean = otherObject match {case _: Point3D => true; case _ => false}
```

Comment:
Isn't this caught by scala style?

Reply:
Scalastyle is not catching it for me. I can rewrite `canEqual` as `otherObject.isInstanceOf[Point3D]`, would that be ok?

Reply:
No, it is fine.
I was now thinking if we wanted Point3D(1,1,0)==Point2D(1,1) to be true or not: it might be reasonable but if this wasn't so before, that's fine.